File: 22- Jump to last index/C++/solution3.cpp
// By keeping track of the max index that we can reach:

// Time complexity: O(n)
// Space complexity: O(1)

bool canJump(vector<int> arr){
  int n = arr.size();
  int maxIndex = 0;
  for(int i = 0; i < n; i++){
    if(i > maxIndex) return false;
    else maxIndex = max(maxIndex, i+arr[i]);
  }
  return maxIndex >= n-1;
}

File: 22- Jump to last index/Python/solution3.py
# By keeping track of the max index that we can reach:

# Time complexity: O(n)
# Space complexity: O(1)


def canJump(arr):
  n = len(arr)
  maxIndex = 0
  for i in range(n):
    if i > maxIndex:
      return False
    else:
      maxIndex = max(maxIndex, i+arr[i])
  return maxIndex >= n-1


